ShippingApp/src/main/java/com/javaschool/queue/QueueSender.java

```diff
@@ -0,0 +1,105 @@
+package com.javaschool.queue;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.javaschool.common.PackageSize;
+import com.javaschool.common.PackageType;
+import com.javaschool.common.QueueException;
+import com.javaschool.common.QueueMessageRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.amqp.core.AmqpTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class QueueSender {
+
+    private static final Logger logger = LoggerFactory.getLogger(QueueSender.class);
+
+    private AmqpTemplate rabbitTemplate;
+
+    @Value("${config-rabbit.routing-key}")
```

Comment:
using @Value like this can quickly become problematic. Let's say you change the entry name in the YAML/properties file, you would break the code each place where this value is being read. Use encapsulation instead, to avoid these problems, by isolating this value the client code will see it through an abstraction (hence not know about the concrete implementation/actual entry name), it would be type safe and if you change the entry name you would only need to modify it in one place or even not at all.

Reply:
The values were changed to @ConfigurationProperties.